### PR TITLE
refactor(cli): update lambda custom resource nodejs runtime version

### PR DESCRIPTION
**Pull request by UnleashedMind**
re #2999
re #2617 

*Issue #, if available:*

*Description of changes:*
This PR updates the nodejs runtime version, from nodejs8.10 to nodejs10.x,  for all the Lambda backed custom resources in the Cloudformation templates used by the Amplify CLI. 
This is done according to the Lambda runtime support's [deprecation schdule](https://docs.aws.amazon.com/lambda/latest/dg/runtime-support-policy.html)

There's also logic to automatically check and update the nodejs versions for projects initialized by Amplify CLI prior to this PR update. So the customers do not need to manually update the versions. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/project-config-version-check.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { Context } from './domain/context';
+import inquirer from './domain/inquirer-helper';
+
+const prevLambdaRuntimeVersion = 'nodejs8.10';
```
Comment (kaustavghosh06) [inline packages/amplify-cli/src/project-config-version-check.ts]: Shouldn't this be an array - to be extensible for future migrations?

**Comment by kaustavghosh06** [comment]
Please add a check for --yes param as well for CI/CD systems.

File: packages/amplify-category-analytics/provider-utils/awscloudformation/cloudformation-templates/pinpoint-cloudformation-template.json [hunk 2]
```diff
@@ -1,428 +1,368 @@
 {
```
[the remark this answers is not in the document]
Comment (UnleashedMind) [reply packages/amplify-category-analytics/provider-utils/awscloudformation/cloudformation-templates/pinpoint-cloudformation-template.json]: prettier

File: packages/amplify-provider-awscloudformation/lib/update-idp-roles-cfn.json [hunk 3]
```diff
@@ -1,172 +1,145 @@
 {
-
```
[the remark this answers is not in the document]
Comment (UnleashedMind) [reply packages/amplify-provider-awscloudformation/lib/update-idp-roles-cfn.json]: prettier

File: packages/amplify-cli/src/project-config-version-check.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+          return;
+        }
+
+        const fileNames = fs.readdirSync(resourceDirPath);
+        fileNames.forEach(fileName => {
+          const filePath = path.join(resourceDirPath, fileName);
+          if (!fs.statSync(filePath).isFile()) {
+            return;
+          }
+          const templateFileNamePattern = new RegExp('cloudformation-template');
```
[the remark this answers is not in the document]
Comment (UnleashedMind) [reply packages/amplify-cli/src/project-config-version-check.ts]: changed

**Comment by kekkler** [comment]
https://github.com/aws-amplify/amplify-cli/issues/3083

**Pull request merged**